File: ruby/bob/bob.rb
module BookKeeping
  VERSION = 1 # Where the version number matches the one in the test.
end


class Bob 

    def self.hey remark
      
        if remark.scan(/[A-Z]/).size > 1 && remark.scan(/[a-z]/).size == 0
          "Whoa, chill out!" 
        elsif remark.strip[-1] == "?" 
          "Sure." 
        elsif remark.scan(/\s/).size == remark.size 
          "Fine. Be that way!"
        else  
          "Whatever."
        end  
    end
    
end

File: javascript/bob/bob.js
var Bob = function() {
 
};

Bob.prototype.hey = (remark) => {
    let size = (x) => x === null ? 0 : x.length; 

    if (size(remark.match(/\s/g)) == size(remark)) {
        return 'Fine. Be that way!';
    } else if (
        size(remark.match(/[A-Z]/g)) > 1 &&
		remark.match(/[a-z]/g) === null
    ) {
        return 'Whoa, chill out!';
    } else if (remark.trim().endsWith('?')) { //es6 
        return 'Sure.';
    } else {
        return 'Whatever.';
    }
};

module.exports = Bob;